### PR TITLE
Fix product-microgateway issue 123: Remove duplicate code

### PR DESCRIPTION
**Pull request by Rajith90**

### CODE REVIEW

File: components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal [hunk 1]
```diff
@@ -258,6 +162,53 @@ public function OAuthAuthProvider::authenticate (APIRequestMetaDataDto apiReques
     return apiKeyValidationDto;
 }
 
+public function OAuthAuthProvider::invokeKeyValidation(APIRequestMetaDataDto apiRequestMetaDataDto) returns (boolean,
```
Comment (isurulucky) [inline components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal]: Add doc comment since this is a public method.

Comment (malinthaprasan) [reply components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal]: I will add this while incorpertating https://github.com/wso2/product-microgateway/pull/144 with changes in this PR

Comment (malinthaprasan) [reply components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal]: My bad. Forgot to add it in 144. Will add it in my next PR.

Comment (malinthaprasan) [reply components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal]: Added with https://github.com/wso2/product-microgateway/pull/158

Comment (Rajith90) [reply components/micro-gateway-core/src/main/ballerina/gateway/oauthn_handler.bal]: Thanks @malinthaprasan for refactoring based on comments

**Pull request merged**